ssl/statem/extensions_srvr.c

```diff
@@ -1645,12 +1645,22 @@ EXT_RETURN tls_construct_stoc_key_share(SSL_CONNECTION *s, WPACKET *pkt,
         }
         return EXT_RETURN_NOT_SENT;
     }
-    if (s->hit && (s->ext.psk_kex_mode & TLSEXT_KEX_MODE_FLAG_KE_DHE) == 0) {
-        /*
-         * PSK ('hit') and explicitly not doing DHE (if the client sent the
-         * DHE option we always take it); don't send key share.
+    if (s->hit) {
+        /* PSK ('hit') */
+
+        /* If we're doing PSK ('hit') but the client doesn't support psk-dhe,
```

Comment:
Nit: Multiline comments need to start with `/*` and end with `*/` on separate lines.